pkg/processor/spec_processor.go

```diff
@@ -126,7 +126,13 @@ func (sp *SpecProcessor) processMatch(selector string, primitivesOnly bool) (int
 	if _, allowed := sp.allowedResources[objName]; !allowed {
 		return nil, fmt.Errorf("references can only point at direct dependencies: %s", selector)
 	}
-	fieldValue := resources.GetNestedField(res.Object, names[1:]...)
+	// To avoid overcomplicated format of reference like this: $((res1#{$.a.string}))
+	// And have something like this instead: $((res1#a.string))
+	jsonPath := fmt.Sprintf("{$.%s}", parts[1])
```

Comment:
Initially I added support of full JsonPath, but it looked really ugly and unreadable: `$((res1#{$.a.string}))`.
Thus I added a shortened format support : `$((res1#a.string))`.
If we find some limitations (for example, some use cases where `$.` prefix is not appropriate), we can add a fallback logic (for example, don't add any prefix/suffix for the expressions starting with bracket `{`).